### DISCUSSION TITLE
Cannot connect to PNPOnline with Token or Thumbprint. Wanting to set up for unattended use without using legacy auth.

### DISCUSSION THREAD

**Discussion by uiklon** [pnp/powershell#3223] [Q&A]
Hi, 

I have tried using tokens and Thumbprint. It only works for my machine with Thumbprint. I'm wanting to have it so i can run as System on a server and edit SharePoint files. 

When i try a accesstoken it comes with access denied. Thumbprint works on my account but not other accounts.

What is the best way to create an Azure app and connect with edit permissions on a site but allow multiple accounts to run it. 

This code lets me connect but everything i've tried is access denied. 

#Authenticate and connect to Microsoft Graph
$tokenBody = @{
  Grant_Type = "client_credentials"
  Scope = "https://graph.microsoft.com/.default"
  Client_Id = '######'
  Client_Secret = '#######
}

$SiteURL = "https://#####.sharepoint.com/sites/######"

$tokenResponse = Invoke-RestMethod -Uri "https://login.microsoftonline.com/#######/oauth2/v2.0/token" -Method POST -Body $tokenBody

Connect-PnPOnline -Url $SiteURL -AccessToken $tokenResponse.access_token

Get-pnpweb
Error is: "Get-pnpweb : The remote server returned an error: (401) Unauthorized"

**Comment by yesh-rajawat-robomq** [comment]
Any update on this?
